crates/eww/src/widgets/graph.rs

```diff
@@ -0,0 +1,159 @@
+    @extends gtk::Bin, gtk::Container, gtk::Widget;
+}
+
+// wrapper! { pub struct Graph(ObjectSubclass<GraphPriv>) @extends gtk::Box, gtk::Container, gtk::Widget; }
+
+#[derive(Default)]
+pub struct GraphPriv {
+    value: RefCell<f64>,
+    thickness: RefCell<f64>,
+    range: RefCell<u64>,
+    history: RefCell<VecDeque<(std::time::Instant, f64)>>,
+    content: RefCell<Option<gtk::Widget>>,
+}
+
+fn update_history(graph: &GraphPriv, v: (std::time::Instant, f64)) {
+    let mut history = graph.history.borrow_mut();
+    history.push_back(v);
+    while let Some(entry) = history.front() {
+        if std::time::Instant::now().duration_since(entry.0).as_millis() as u64 > *graph.range.borrow() {
+            history.pop_front();
+        }
+        else {
+            break
+        }
+    }
+}
+
+impl ObjectImpl for GraphPriv {
+    // glib_object_impl!();
+    fn properties() -> &'static [glib::ParamSpec] {
+        use once_cell::sync::Lazy;
+        static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+            vec![
+                glib::ParamSpec::new_double("value", "Value", "The value", 0f64, 100f64, 0f64, glib::ParamFlags::READWRITE),
+                glib::ParamSpec::new_double("thickness", "Thickness", "The Thickness", 0f64, 100f64, 0f64, glib::ParamFlags::READWRITE),
+                glib::ParamSpec::new_uint64("range", "Range", "The Range", 0u64, u64::MAX, 10u64, glib::ParamFlags::READWRITE),
+            ]
+        });
+
+        PROPERTIES.as_ref()
+    }
+
+    //    fn constructed(&self, obj: &Self::Type) {
+    //        self.parent_constructed(obj);
+    //    }
+
+    fn set_property(&self, obj: &Self::Type, _id: usize, value: &glib::Value, pspec: &glib::ParamSpec) {
+        match pspec.name() {
+            "value" => {
+                self.value.replace(value.get().unwrap());
+                update_history(self, (std::time::Instant::now(), value.get().unwrap()));
+                obj.queue_draw();
+            }
+            "thickness" => {
+                self.thickness.replace(value.get().unwrap());
+            }
+            "range" => {
+                self.range.replace(value.get().unwrap());
+            }
+
+            x => panic!("Tried to set inexistant property of G()raph: {}", x,),
+        }
+    }
+
+    fn property(&self, _obj: &Self::Type, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
+        match pspec.name() {
+            "value" => self.value.borrow().to_value(),
+            "thickness" => self.thickness.borrow().to_value(),
+            "range" => self.range.borrow().to_value(),
+            x => panic!("Tried to access inexistant property of Graph: {}", x,),
+        }
+    }
+}
+
+#[object_subclass]
+impl ObjectSubclass for GraphPriv {
+    type ParentType = gtk::Bin;
+    type Type = Graph;
+
+    const NAME: &'static str = "Graph";
+}
+
+impl Graph {
+    pub fn new() -> Self {
+        glib::Object::new::<Self>(&[]).expect("Failed to create MyAwesome Widget")
```

Comment:
```suggestion
        glib::Object::new::<Self>(&[]).expect("Failed to create graph Widget")
```
My awesome? Based on context this should be Graph, right?